### ISSUE TITLE
Please give me some advice to reproduce the results in the paper

### ISSUE THREAD

**Issue by LKJacky** [luminolx/ScaleNet#1]
Thanks for your excellent work. I try to reproduce the scalenet-s0. However, I only get the top-1 accuracy of 76.58.
Here is my model and training recipe.

The model is created by the blow code with Timm.

```python
  model_kwargs = dict(
      block_args=decode_arch_def(
          [['ir_r1_k5_s1_e1_c16_se0.25'], 
          ['ir_r1_k5_s2_e6_c32_se0.25'], 
          ['ir_r2_k7_s2_e6_c40_se0.25', 'ir_r1_k5_s1_e6_c40_se0.25', 'ir_r1_k3_s1_e6_c40_se0.25'],
          ['ir_r2_k5_s2_e6_c80_se0.25'], 
          ['ir_r2_k7_s1_e6_c96_se0.25'], 
          ['ir_r1_k7_s2_e6_c192_se0.25', 'ir_r1_k5_s1_e6_c192_se0.25', 'ir_r1_k7_s1_e6_c192_se0.25'],
          ['ir_r1_k5_s1_e6_c320_se0.25'],]
          , depth_multiplier=1.0, group_size=None),
      num_features=round_chs_fn(1280),
      stem_size=32,
      round_chs_fn=round_chs_fn,
      act_layer=resolve_act_layer(kwargs, 'hard_swish'),
      norm_layer=kwargs.pop('norm_layer', None)
      or partial(nn.BatchNorm2d, **resolve_bn_args(kwargs)),
      **kwargs,
  )
  model = _create_effnet(variant, pretrained, **model_kwargs)
```

The training recipe is based on Timm.
```shell
sh ./dist_timm_train.sh 8 data/imagenet_torch --model scale_net_timm -b 128 --sched step --epochs 300 --decay-epochs 2.4 --decay-rate .97 --opt rmsproptf --opt-eps .001 -j 8 --warmup-lr 1e-6 --weight-decay 1e-5 --drop 0.2 --drop-path 0.2 --model-ema --model-ema-decay 0.9999 --aa rand-m9-mstd0.5 --remode pixel --reprob 0.2 --amp --lr 0.08 --warmup-epochs=3 
```

Please give me some advice to reproduce the results in the paper.

**Comment by luminolx** [comment]
We set batch size as 1024 and used sync-batchnorm layers. Meanwhile, the block args are not fully consistent to our scalenet-s0 model, please find them in fig10 of the supplementary materials.

**Comment by LKJacky** [comment]
We train the model on 8 GPUs, so the total batch size is 1024=8*128.
It's my fault that I only use regular BatchNorm layers.
I recheck the block args. The only difference is the expand ratio of the first block. In the paper, the expand ratio is 1 when searching, but 6 in fig10. If we use expand ratio=6 as fig10, the flops and params will exceed that in the paper.
I would appreciate it very much if you provided your submodel code and retraining recipe.

**Comment by luminolx** [comment]
We used a self-implemented function for counting flops, see line 665 in core/sampler/evolution/evolution_sampler.py. I have no idea with if there is mismatching between yours and ours. We have applied this function computing EfficientNets' flops for checking our correctness.
Our submodel code is same to that in supernet training and sampling, see core/search_space/model_initializer.py and core/search_space/ops.py. You can find model config in config/training_super_supernet.yaml. Except this, please mention that our S0 model has [1, 1, 4, 2, 2, 3, 1] blocks in each stages, but yours is [1, 1, 3, 1, 1, 3, 1]. They are different a lot. I have no idea with if you updated correct block args.
Also, the retraining recipe is presented in "Model Retraining" part, Section A of Supplementart Material.

**Comment by LKJacky** [comment]
Thanks for your help. 

My question on the model structure may be solved. 
There may be a bug in the SqueezeExcite in the repo. I export sub-models based on the official code. The numbers of the sub-model parameters are much larger than that in the paper. When I set the reduce=expand_ratio*4 (please refer to the Efficientnet in timm) from reduce=4, the number of parameters matches that in the paper.

We export the sub-models with reduce=expand_ratio*4 and count their flops and params using thop and ptflops. They basically match the paper.
```
simple format: {'net': {1: [5], 2: [5], 3: [7, 7, 5, 3], 4: [5, 5], 5: [7, 7], 6: [7, 5, 7], 7: [5]}, 'reso': 224, 'c_ratio': 1.0}
official format: [0, 2, 0, 2, 0, 0, 0, 0, 0, 0, 3, 3, 2, 1, 0, 0, 0, 2, 2, 0, 0, 0, 0, 0, 3, 3, 0, 0, 0, 0, 0, 3, 2, 3, 0, 0, 0, 0, 2, 0, 0, 0, 0]
official flops 352.69088
thop: 377.190592 4.442245
ptflops: 364.98 MMac 4.44 M

simple format: {'net': {1: [5, 5], 2: [5, 5], 3: [7, 7, 5, 3, 3], 4: [5, 5, 5], 5: [7, 7, 7], 6: [7, 5, 7, 7], 7: [5, 5]}, 'reso': 256, 'c_ratio': 1.04}
official format: [0, 2, 2, 2, 2, 0, 0, 0, 0, 0, 3, 3, 2, 1, 1, 0, 0, 2, 2, 2, 0, 0, 0, 0, 3, 3, 3, 0, 0, 0, 0, 3, 2, 3, 3, 0, 0, 0, 2, 2, 0, 0, 0]
official flops 800.295584
thop: 849.989024 7.431683
ptflops: 825.21 MMac 7.43 M

simple format: {'net': {1: [5, 5], 2: [5, 5], 3: [7, 7, 5, 3, 3, 3], 4: [5, 5, 5], 5: [7, 7, 7], 6: [7, 5, 7, 7, 7], 7: [5, 5]}, 'reso': 304, 'c_ratio': 1.2}
official format: [0, 2, 2, 2, 2, 0, 0, 0, 0, 0, 3, 3, 2, 1, 1, 1, 0, 2, 2, 2, 0, 0, 0, 0, 3, 3, 3, 0, 0, 0, 0, 3, 2, 3, 3, 3, 0, 0, 2, 2, 0, 0, 0]
official flops 1504.197058
thop: 1708.377754 10.21318
ptflops: 1.67 GMac 10.21 M

simple format: {'net': {1: [5, 5], 2: [5, 5], 3: [7, 7, 5, 3, 3, 3], 4: [5, 5, 5], 5: [7, 7, 7], 6: [7, 5, 7, 7, 7], 7: [5, 5]}, 'reso': 354, 'c_ratio': 1.4}
official format: [0, 2, 2, 2, 2, 0, 0, 0, 0, 0, 3, 3, 2, 1, 1, 1, 0, 2, 2, 2, 0, 0, 0, 0, 3, 3, 3, 0, 0, 0, 0, 3, 2, 3, 3, 3, 0, 0, 2, 2, 0, 0, 0]
official flops 2756.158257
thop: 3173.013021 13.198492
ptflops: 3.1 GMac 13.2 M
```

Besides, our block setting is correct. Our implementation has  [1, 1, 4, 2, 2, 3, 1]  blocks. As 'r2' in our block_args means the block will repeat twice, like 'ir_r2_k7_s2_e6_c40_se0.25' in stage 3.

If reduce=expand_ratio*4 is right. The only difference is in the training arguments.
I'm curious about the training arguments except that mentioned in the Supplementart Material, like dropout, droppath, and so on.

**Comment by luminolx** [comment]
I remember this bug which was in EfficientNet's code or another work, we followed it in ours.
